File: Inventor-Training-Material-0c232f96f36e1e0bed31b07a6a6ebb901a31dd40/Module 20 - Browser/Browser.VB.NET/BrowserSample/UserControl1.vb
Imports Inventor



Public Class UserControl1

    Private Sub Button1_Click(ByVal sender As System.Object, ByVal e As System.EventArgs) Handles Button1.Click
        MsgBox("The control in my pane is clicked !")
    End Sub

    Public Sub DrawASketchRectangle(ByVal InvApp As Inventor.Application)

        Dim oPartDef As PartComponentDefinition = InvApp.ActiveDocument.ComponentDefinition

        Dim oPoint1 As Point2d = InvApp.TransientGeometry.CreatePoint2d(0, 0)
        Dim oPoint2 As Point2d = InvApp.TransientGeometry.CreatePoint2d(10, 10)

        oPartDef.Sketches(1).SketchLines.AddAsTwoPointRectangle(oPoint1, oPoint2)

        InvApp.ActiveDocument.Views(1).Fit()

    End Sub

End Class

File: testProjectLpzIgnore/testProjectLpzIgnore/UserControl1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Drawing;
using System.Data;
using System.Text;
using System.Windows.Forms;
using Inventor;


namespace BrowserSample
{
    public partial class UserControl1 : UserControl
    {
        public UserControl1()
        {
            InitializeComponent();
        }

        private void button1_Click(object sender, EventArgs e)
        {
            MessageBox.Show("The control in my pane is clicked !");
        }
        public void DrawASketchRectangle(Inventor.Application InvApp)
        {
            PartDocument oPartDoc = InvApp.ActiveDocument as PartDocument;
            PartComponentDefinition oPartDef = oPartDoc.ComponentDefinition;

            Point2d oPoint1 = InvApp.TransientGeometry.CreatePoint2d(0, 0);
            Point2d oPoint2 = InvApp.TransientGeometry.CreatePoint2d(10, 10);

            oPartDef.Sketches[1].SketchLines.AddAsTwoPointRectangle(oPoint1, oPoint2);

            InvApp.ActiveDocument.Views[1].Fit();

        }


    }
}
